apps/frontend-v3/app/(app)/pools/[chain]/[variant]/[id]/page.tsx

```diff
@@ -1,10 +1,13 @@
 import { PoolDetail } from '@repo/lib/modules/pool/PoolDetail/PoolDetail'
 import { TransactionStateProvider } from '@repo/lib/modules/transactions/transaction-steps/TransactionStateProvider'
+import { VebalLockInfoProvider } from '@repo/lib/modules/vebal/lock/VebalLockInfoProvider'
 
 export default function PoolPage() {
   return (
     <TransactionStateProvider>
-      <PoolDetail />
+      <VebalLockInfoProvider>
+        <PoolDetail />
+      </VebalLockInfoProvider>
```

Comment:
I'm worried about this being on every pool page. Is it needed on every pool?

Reply:
Can you link to the change you made? If you need the boost having a provider or some other solution is not going to change much.

Reply:
Ok, but changing from provider to a normal hook doesn't change anything, so what's the benefit of doing that? It may even introduce problems from using it in multiple places?

What happens with this stale time on the vebal page? What if they just updated their lock information but the data is cached for another minute? Will it result in a UI lag where the user sees their old lock information?

Reply:
Because you use the `useVebalLockData` hook on the pool page, right? So everytime the pool page loads the contents of that hook will be run, basically the same as if it was a provider. The only difference with it being in the provider is that you then can be sure you are only instantiating the hook once, even if you use the hook in multiple places on the pool page.